src/Instrumentation/Laravel/_register.php

```diff
@@ -5,14 +5,18 @@
 use OpenTelemetry\Contrib\Instrumentation\Laravel\LaravelInstrumentation;
 use OpenTelemetry\SDK\Sdk;
 
-if (class_exists(Sdk::class) && Sdk::isInstrumentationDisabled(LaravelInstrumentation::NAME) === true) {
-    return;
-}
+(function () {
+    $instrumentation = new LaravelInstrumentation();
 
-if (extension_loaded('opentelemetry') === false) {
-    trigger_error('The opentelemetry extension must be loaded in order to autoload the OpenTelemetry Laravel auto-instrumentation', E_USER_WARNING);
+    if (class_exists(Sdk::class) && Sdk::isInstrumentationDisabled($instrumentation->getName()) === true) {
+        return;
+    }
 
-    return;
-}
+    if (extension_loaded('opentelemetry') === false) {
+        trigger_error('The opentelemetry extension must be loaded in order to autoload the OpenTelemetry Laravel auto-instrumentation', E_USER_WARNING);
 
-LaravelInstrumentation::register();
+        return;
+    }
+
+    $instrumentation->activate();
```

Comment:
Could there be a way to register this instrument with the API so that activation is delegated elsewhere?

Reply:
Toying with this idea as part of this PR now, so resolving this comment.